lib/view_component/base.rb

```diff
@@ -140,6 +145,55 @@ def with(area, content = nil, &block)
       nil
     end
 
+    # Build a Slot instance on a component,
+    # exposing it for use inside the
+    # component template.
+    #
+    # slot: Name of Slot, in symbol form
+    # **args: Arguments to be passed to Slot initializer
+    #
+    # For example:
+    # <%= render(SlotsComponent.new) do |component| %>
+    #   <% component.slot(:footer, class_names: "footer-class") do %>
+    #     <p>This is my footer!</p>
+    #   <% end %>
+    # <% end %>
+    #
+    def slot(slot_name, **args, &block)
+      # Raise ArgumentError if `slot` does not exist
+      unless slots.keys.include?(slot_name)
```

Comment:
Should this only raise in dev/test?

Reply:
That's a fair question. My concern is that if this doesn't fail in prod, what should the behavior be in this case?

Reply:
>  what should the behavior be in that situation?

Yeah, I think it's a bit tricky since raising is the best way to bring attention to the problem, but at the cost of a 500 when the slot might not be critical or was just a typo. 🤔

Reply:
@BlakeWilliams what would you recommend the behavior to be? Have it not render?

Reply:
I think not rendering would be ideal in most cases. Unless a slot is critical, I'd much rather render the page with a missing icon or some missing text than have the page be entirely unaccessible.